pkg/util/util_unix.go

```diff
@@ -88,11 +87,6 @@ func TouchFile(filePath string) error {
 		return err
 	}
 	if !exists {
-		err := os.MkdirAll(filepath.Dir(filePath), 0755)
```

Comment:
Is this assumption true for all csi drivers?

Reply:
The path for the temp registration file is hidden from the users, the steps to create it are:

- take the kubelet registration path: `/var/lib/kubelet/plugins/<drivername.example.com>/csi.sock`
- on a successful registration, write the temp file to the same dirname: `/var/lib/kubelet/plugins/<drivername.example.com>/registration`

I think the directory `/var/lib/kubelet/plugins/<drivername.example.com>/` is created in advance in the driver spec e.g. from the README

```
      volumes:
        - name: plugin-dir
          hostPath:
            path: /var/lib/kubelet/plugins/<drivername.example.com>/
            type: DirectoryOrCreate
```

I checked a few drivers:

- [aws ebs](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/4b64c4e1b1a252c94e5940bcc588c9233c03f3d1/deploy/kubernetes/base/node.yaml#L113)
- [pd csi](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/7a126c25d05ffe04feecf950e0b0fb89e4d7af67/deploy/kubernetes/base/node_linux/node.yaml#L79)
- [azuredisk](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/7b3b87cb3420f0051c128495ea2a01bb7e1ddf38/deploy/v1.21.0/csi-azuredisk-node.yaml#L144)

I think the assumption that this directory is created is true therefore node-driver-registrar should only create the file assuming that the dir already exists